draft-thoji-scone-trone-protocol.md

```diff
@@ -310,6 +310,109 @@ This transport parameter is valid for QUIC versions 1 {{QUIC}} and 2
 transport parameters, and frame types registries established in {{Sections 22.2,
 22.3, and 22.4 of QUIC}}.
 
+# TRONE Indication {#trone-indication}
+
+QUIC endpoints can signal potential support for TRONE before the completion of
+the QUIC handshake by attaching a TRONE indication packet after the last QUIC
+long header packet in the first UDP datagram. The TRONE indication provides an
+opportunistic signal to network elements that the client might support TRONE.
+Network elements can use this as an early hint, but must await confirmation of
+TRONE support by observing a full TRONE packet after the handshake completes.
+
+When sending QUIC short header packets in the first flight, endpoints SHOULD
+send them separately from the first datagram carrying the TRONE indication
+packet, to avoid receivers incapable of decoding TRONE packets from dropping
+the short header packet.
+
+## TRONE Indication Packet
+
+A TRONE indication packet has a format with fully reversed field order compared
+to TRONE packets to facilitate detection by parsing from the end of the packet.
+
+{{fig-trone-indication-packet}} shows the format of the TRONE indication packet:
+
+~~~ artwork
+TRONE Indication Packet {
+  Source Connection ID (0..2040),
+  Source Connection ID Length (8),
+  Destination Connection ID (0..2040),
+  Destination Connection ID Length (8),
+  Version (32) = 0xTRONE1 or 0xTRONE2,
+  Rate Signal (6) = 0x3F,
+  Reserved (1),
+  Header Form (1) = 1
+}
+~~~
+{: #fig-trone-indication-packet title="TRONE Indication Packet Format"}
+
+A TRONE indication packet is attached directly after a QUIC Initial packet in
+the first UDP datagram. The Rate Signal field MUST be set to 0x3F (63) when
+sent as a TRONE indication packet.
+
+## Sending TRONE Indication
+
+A QUIC endpoint that supports TRONE MAY attach a TRONE indication packet after
+a QUIC Initial packet in the first UDP datagram. This is an opportunistic
+signal that is not guaranteed to be processed by the receiver, as the receiver
+has not yet confirmed its willingness to receive TRONE packets.
+
+When attaching a TRONE indication, the endpoint SHOULD:
+* Place it immediately after the QUIC Initial packet in the datagram
```

Comment:
Is this consistent with a Datagram that has a coalesced 0RTT packet, or Mozilla-like zero padding? The parsing rules indicate this should be the very last thing.